deploy/examples/create-external-cluster-resources.py

```diff
@@ -1731,24 +1765,25 @@ def gen_json_out(self):
                     },
                 }
             )
-        if self.out_map["RBD_METADATA_EC_POOL_NAME"]:
+        # if 'RADOS_NAMESPACE' exists, then only add the "RADOS_NAMESPACE" namespace
+        if (
+            self.out_map["RADOS_NAMESPACE"]
+            and self.out_map["RESTRICTED_AUTH_PERMISSION"]
+            and not self.out_map["RBD_METADATA_EC_POOL_NAME"]
+        ):
             json_out.append(
                 {
-                    "name": "ceph-rbd",
-                    "kind": "StorageClass",
+                    "name": "rados-namespace",
+                    "kind": "CephBlockPoolRadosNamespace",
                     "data": {
-                        "dataPool": self.out_map["RBD_POOL_NAME"],
-                        "pool": self.out_map["RBD_METADATA_EC_POOL_NAME"],
-                        "csi.storage.k8s.io/provisioner-secret-name": f"rook-{self.out_map['CSI_RBD_PROVISIONER_SECRET_NAME']}",
-                        "csi.storage.k8s.io/controller-expand-secret-name": f"rook-{self.out_map['CSI_RBD_PROVISIONER_SECRET_NAME']}",
-                        "csi.storage.k8s.io/node-stage-secret-name": f"rook-{self.out_map['CSI_RBD_NODE_SECRET_NAME']}",
+                        "radosNamespaceName": self.out_map["RADOS_NAMESPACE"],
```

Comment:
this was renamed, right?
```suggestion
                        "radosNamespace": self.out_map["RADOS_NAMESPACE"],
```

Reply:
This is just the the internal parameter for usage  in ocs operator,
actuall flag is kept same `--RADOS-NAMESPACE`

Reply:
Actually, shouldn't this correspond to the `name` field in the rados namespace CR that is being added in #13270?
```suggestion
                        "name": self.out_map["RADOS_NAMESPACE"],
```

Reply:
Or maybe I'm forgetting how the json is used. Does something else translate this JSON into the rados namespace CR?

Reply:
So we are not supporting multiple radosnamesapce within the same k8s for this release,
And this is been consumed here `https://github.com/red-hat-storage/ocs-operator/pull/2267/files#diff-2249436658af47e06252d313cae1379923a6880d9b6ac9bad8d400fb9a433094` line `356` which is just internal code flow

Reply:
Ok I guess it's just confusing that this should still be called `radosNamespaceName`, while we call it `radosNamespace` or `RADOS_NAMESPACE` other places.

Reply:
Agreed. Naming this `radosNamespace` will follow the same trend as `dataPool` and `pool` above.

Reply:
Ok let's keep it since there's an external consumer of this property name.